operator/controllers/resources/role_controller_test.go

```diff
@@ -197,6 +326,20 @@ func TestRoleUpdate(t *testing.T) {
 	})
 }
 
+func setupKubernetesAndTeleport(t *testing.T) (context.Context, auth.ClientI, kclient.Client, *v1.Namespace) {
```

Comment:
Let's wrap the response in a struct, so each component has a name.

Returning a context also looks a bit like a smell - I'd either keep it entirely within the function or get it as a parameter.

Reply:
Done, and extended to user tests as well.